util/pytest_log_filter.py

```diff
@@ -0,0 +1,89 @@
+
+    Example of filtering:
+    This function expects a pytest test report of single test file as input, a string such as the example below:
+
+        ============================= test session starts ==============================
+        platform linux -- Python 3.6.9, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python3
+        cachedir: ../../../../.cache
+        rootdir: /home/kleijwegt/development/odemis, inifile:
+        collecting ... collected 9 items
+
+        comp_canvas_test.py::TestDblMicroscopeCanvas::test_basic_display PASSED  [ 11%]
+        comp_canvas_test.py::TestDblMicroscopeCanvas::test_basic_move PASSED     [ 22%]
+        comp_canvas_test.py::TestDblMicroscopeCanvas::test_conversion_functions PASSED [ 33%]
+        comp_canvas_test.py::TestDblMicroscopeCanvas::test_conversion_methods PASSED [ 44%]
+        comp_canvas_test.py::TestDblMicroscopeCanvas::test_crosshair PASSED      [ 55%]
+        comp_canvas_test.py::TestDblMicroscopeCanvas::test_pyramidal_3x2 FAILED  [ 66%]
+        comp_canvas_test.py::TestDblMicroscopeCanvas::test_pyramidal_one_tile FAILED [ 77%]
+        comp_canvas_test.py::TestDblMicroscopeCanvas::test_pyramidal_zoom FAILED [ 88%]
+        comp_canvas_test.py::TestDblMicroscopeCanvas::test_zoom_move PASSED      [100%]
+
+        =================================== FAILURES ===================================
+        /home/kleijwegt/development/odemis/src/odemis/gui/test/comp_canvas_test.py:742: AssertionError: Tuples differ: (0, 128, 127) != (0, 76, 179)
+        /home/kleijwegt/development/odemis/src/odemis/gui/test/comp_canvas_test.py:501: AssertionError: Tuples differ: (0, 128, 127) != (0, 76, 179)
+        /home/kleijwegt/development/odemis/src/odemis/gui/test/comp_canvas_test.py:604: AssertionError: Tuples differ: (0, 127, 128) != (0, 179, 76)
+        ====================== 3 failed, 6 passed in 7.43 seconds ======================
+
+    For a full test report only failures, errors etc. are of interest, therefore only this part is outputted, the rest is
+    filtered. An example of such an output string is the following example:
+
+        =================================== FAILURES ===================================
+        /home/kleijwegt/development/odemis/src/odemis/gui/test/comp_canvas_test.py:742: AssertionError: Tuples differ: (0, 128, 127) != (0, 76, 179)
+        /home/kleijwegt/development/odemis/src/odemis/gui/test/comp_canvas_test.py:501: AssertionError: Tuples differ: (0, 128, 127) != (0, 76, 179)
+        /home/kleijwegt/development/odemis/src/odemis/gui/test/comp_canvas_test.py:604: AssertionError: Tuples differ: (0, 127, 128) != (0, 179, 76)
+        ====================== 3 failed, 6 passed in 7.43 seconds ======================
+
+    """
+    test_results = re.search("\n\n===(.+?)((.|\n)*)==", log_txt).group()  # Filter to only get the usefull parts of the test report
+    test_results = test_results[test_results.find("="):]  # Remove preceding empty lines
```

Comment:
If you just want to remove empty lines, then you can write simply:
test_results = test_results.lstrip("\n")